### PR TITLE
database creation

### PR DESCRIPTION
**Pull request by anishaf92**

### CODE REVIEW

**Review by vigneshsudhakar — CHANGES_REQUESTED** [review]
_pycache_/
venv/
.idea/

please add these 3 to the .gitignore file and remove .idea folder which you have pushed to this branch

File: models.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    return db
+
+
+def create_table(app):
+    engine = create_engine(app.config['SQLALCHEMY_DATABASE_URI'])
+    db.metadata.create_all(engine)
+    return engine
+
+
+cart = db.Table('cart',
```
Comment (vigneshsudhakar) [inline models.py]: Why is this different from the others?
Please maintain uniformity.

File: models.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                db.Column('user_id', db.Integer, db.ForeignKey('user.id')),
+                db.Column('product_id', db.Integer, db.ForeignKey('product.id'))
+                )
+
+
+class User(db.Model):
+    __tablename__ = "user"
```
Comment (vigneshsudhakar) [inline models.py]: " __tablename__ = "user""

Why is this required in every class?

File: models.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    id = db.Column(db.Integer, primary_key=True)
+    firebase_uid = db.Column(db.String(200), unique=True, nullable=False)
+    username = db.Column(db.String(80), unique=True, nullable=False)
+    email = db.Column(db.String(120), unique=True, nullable=False)
+    mobile = db.Column(db.String(200), unique=True, nullable=False)
+    items_in_cart = db.relationship('Product', secondary=cart, backref='cart-products')
```
Comment (vigneshsudhakar) [inline models.py]: Please add "to_json" function to each class and return a json object containing all the data

Comment (vigneshsudhakar) [inline models.py]: is_active = db.Column(db.Boolean, default=True, nullable=False)
is_deleted = db.Column(db.Boolean, default=False, nullable=False)
created_by = db.Column(db.Integer, nullable=True)
created_date = db.Column(db.DateTime, default=datetime.datetime.utcnow, nullable=False)
updated_by = db.Column(db.Integer, nullable=True)
updated_date = db.Column(db.DateTime, default=datetime.datetime.utcnow, nullable=False)



Add these above 6 fields to all the tables

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,13 @@
+click==8.1.3
```
Comment (vigneshsudhakar) [inline requirements.txt:2]: Why are all these packages added here?
Are they needed?
I see you using just Flask, mysql-connector, dotenv and SQLAlchemy

**Comment by anishaf92** [comment]
> _pycache_/ venv/ .idea/
> 
> please add these 3 to the .gitignore file and remove .idea folder which you have pushed to this branch

**Comment by anishaf92** [comment]
Removed .idea folder and added pycache/ venv/ .idea/ files to gitgnore

> _pycache_/ venv/ .idea/
> 
> please add these 3 to the .gitignore file and remove .idea folder which you have pushed to this branch

Removed .idea folder and added pycache/ venv/ .idea/ files to gitgnore
